lib/core-integration/src/Test/Integration/Scenario/API/Wallets.hs

```diff
@@ -1779,3 +1794,39 @@ spec = do
         ru <- request @ApiWallet ctx ("GET", endpoint) Default newName
         expectResponseCode @IO HTTP.status404 ru
         expectErrorMessage (errMsg404NoWallet wid) ru
+
+    describe "WALLETS_RESYNC_01 - \
+        \ force resync eventually get us back to the same point" $ do
+        scenarioWalletResync01 @'Shelley emptyWallet
+        scenarioWalletResync01 @'Byron emptyRandomWallet
+        scenarioWalletResync01 @'Byron emptyIcarusWallet
```

Comment:
@piotr-iohk this is an example of how I think we could re-structure a bit the integration scenarios. Key points are:

- named function, that can potentially be re-used in other scenarios (imagine for instance 'scenarioWalletRestore' or 'scenarioMakeTransaction' instead of re-writing the same things over and over).

- The scenario itself is generic in terms of wallet type, so that it can easily be replayed with any type of wallet we have.